engines/dfc_provider/app/controllers/dfc_provider/api/base_controller.rb

```diff
@@ -8,7 +8,7 @@ class BaseController < ActionController::Base
 
       before_action :check_authorization,
                     :check_user,
-                    :check_enterprise
+                    :set_enterprise
```

Comment:
```suggestion
                    :current_enterprise
```

And delete the additional method? Otherwise I would probably call it `load_enterprise` to give more context of where it's coming from.

Reply:
Actually, I decided to call this method `check_enterprise` as we are controlling the link between the user and the enterprise at this step, tell me if any issue with that.